SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1172,11 +1163,32 @@ void USpatialActorChannel::UpdateSpatialPosition()
 			return;
 		}
 	}
+	
+	// Check that the Actor has satisfied both minimum thresholds OR either of the maximum thresholds
+	bool bSatisfiesMininumThresholds = false;
+	bool bSatisfiesMaximumThreshold = false;
 
-	// Check that the Actor has moved sufficiently far to be updated
-	const float SpatialPositionThresholdSquared = FMath::Square(GetDefault<USpatialGDKSettings>()->PositionDistanceThreshold);
+	const float TimeSinceLastPositionUpdate = NetDriver->GetElapsedTime() - TimeWhenPositionLastUpdated;
 	FVector ActorSpatialPosition = SpatialGDK::GetActorSpatialPosition(Actor);
-	if (FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate) < SpatialPositionThresholdSquared)
+	const float DistanceTravelledSinceLastUpdateSquared = FVector::DistSquared(ActorSpatialPosition, LastPositionSinceUpdate);
+	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
+
+	const float SpatialMinimumPositionThresholdSquared = FMath::Square(SpatialGDKSettings->MinimumDistanceThreshold);
+
+	if (TimeSinceLastPositionUpdate >= SpatialGDKSettings->MinimumTimeThreshold && DistanceTravelledSinceLastUpdateSquared >= SpatialMinimumPositionThresholdSquared)
+	{
+		bSatisfiesMininumThresholds = true;
+	}
+
+	const float SpatialMaximumPositionThresholdSquared = FMath::Square(SpatialGDKSettings->MaximumDistanceThreshold);
```

Comment:
nit:
I'd prefer to just have a single `bSatisifiedPositionUpdateRequirements` variable, and group your variable definition and logic statements separately. So you get -
// define all variables
// check all conditions separately
if (min)
else if (max1)
else if (max2)